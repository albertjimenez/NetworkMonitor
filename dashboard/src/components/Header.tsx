import {Button, Layout} from 'antd';
import {AppstoreAddOutlined, MenuOutlined,} from '@ant-design/icons';
import {Link} from "react-router-dom";

const {Header: AntHeader} = Layout;


const Header = () => {
    return (
        <AntHeader className="header">
            <div className="logo">
                <Link to="/">
                    <AppstoreAddOutlined style={{fontSize: '24px', color: '#fff'}}/>
                    <span style={{color: '#fff', marginLeft: '10px'}}>Network Monitor</span>
                </Link>
            </div>
            <div style={{float: 'right', display: 'flex', alignItems: 'center'}}>
                <Button
                    icon={<MenuOutlined/>}
                    className="drawer-toggle"
                    style={{display: 'none'}} // Hidden by default
                />
            </div>
        </AntHeader>
    );
};

export default Header;
