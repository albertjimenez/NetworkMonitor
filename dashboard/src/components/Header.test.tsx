import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Header from './Header';
import { describe, it, expect } from 'vitest';
import '@testing-library/jest-dom';

describe('Header component', () => {
    it('renders the logo with "Network Monitor" text', () => {
        render(
            <BrowserRouter>
                <Header />
            </BrowserRouter>
        );

        // Check for logo text "Network Monitor"
        const logoText = screen.getByText(/Network Monitor/i);
        expect(logoText).toBeInTheDocument();
    });

    it('Header matches snapshot', () => {
        const header = render(
            <BrowserRouter>
                <Header />
            </BrowserRouter>
        );

        expect(header).toMatchSnapshot();
    });

});
