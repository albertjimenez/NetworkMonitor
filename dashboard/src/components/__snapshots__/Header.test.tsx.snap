// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component > Header matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="ant-layout-header header css-dev-only-do-not-override-14i19y2"
      >
        <div
          class="logo"
        >
          <a
            href="/"
          >
            <span
              aria-label="appstore-add"
              class="anticon anticon-appstore-add"
              role="img"
              style="font-size: 24px; color: rgb(255, 255, 255);"
            >
              <svg
                aria-hidden="true"
                data-icon="appstore-add"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <defs>
                  <style />
                </defs>
                <path
                  d="M464 144H160c-8.8 0-16 7.2-16 16v304c0 8.8 7.2 16 16 16h304c8.8 0 16-7.2 16-16V160c0-8.8-7.2-16-16-16zm-52 268H212V212h200v200zm452-268H560c-8.8 0-16 7.2-16 16v304c0 8.8 7.2 16 16 16h304c8.8 0 16-7.2 16-16V160c0-8.8-7.2-16-16-16zm-52 268H612V212h200v200zm52 132H560c-8.8 0-16 7.2-16 16v304c0 8.8 7.2 16 16 16h304c8.8 0 16-7.2 16-16V560c0-8.8-7.2-16-16-16zm-52 268H612V612h200v200zM424 712H296V584c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v128H104c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h128v128c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V776h128c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8z"
                />
              </svg>
            </span>
            <span
              style="color: rgb(255, 255, 255); margin-left: 10px;"
            >
              Network Monitor
            </span>
          </a>
        </div>
        <div
          style="float: right; display: flex; align-items: center;"
        >
          <button
            class="ant-btn css-dev-only-do-not-override-14i19y2 ant-btn-default ant-btn-icon-only drawer-toggle"
            style="display: none;"
            type="button"
          >
            <span
              class="ant-btn-icon"
            >
              <span
                aria-label="menu"
                class="anticon anticon-menu"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="menu"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
                  />
                </svg>
              </span>
            </span>
          </button>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="ant-layout-header header css-dev-only-do-not-override-14i19y2"
    >
      <div
        class="logo"
      >
        <a
          href="/"
        >
          <span
            aria-label="appstore-add"
            class="anticon anticon-appstore-add"
            role="img"
            style="font-size: 24px; color: rgb(255, 255, 255);"
          >
            <svg
              aria-hidden="true"
              data-icon="appstore-add"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <defs>
                <style />
              </defs>
              <path
                d="M464 144H160c-8.8 0-16 7.2-16 16v304c0 8.8 7.2 16 16 16h304c8.8 0 16-7.2 16-16V160c0-8.8-7.2-16-16-16zm-52 268H212V212h200v200zm452-268H560c-8.8 0-16 7.2-16 16v304c0 8.8 7.2 16 16 16h304c8.8 0 16-7.2 16-16V160c0-8.8-7.2-16-16-16zm-52 268H612V212h200v200zm52 132H560c-8.8 0-16 7.2-16 16v304c0 8.8 7.2 16 16 16h304c8.8 0 16-7.2 16-16V560c0-8.8-7.2-16-16-16zm-52 268H612V612h200v200zM424 712H296V584c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v128H104c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h128v128c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V776h128c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8z"
              />
            </svg>
          </span>
          <span
            style="color: rgb(255, 255, 255); margin-left: 10px;"
          >
            Network Monitor
          </span>
        </a>
      </div>
      <div
        style="float: right; display: flex; align-items: center;"
      >
        <button
          class="ant-btn css-dev-only-do-not-override-14i19y2 ant-btn-default ant-btn-icon-only drawer-toggle"
          style="display: none;"
          type="button"
        >
          <span
            class="ant-btn-icon"
          >
            <span
              aria-label="menu"
              class="anticon anticon-menu"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="menu"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
                />
              </svg>
            </span>
          </span>
        </button>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
